### DISCUSSION TITLE
Better PR View - expanding the drawer to show a more detailed view of a PR

### DISCUSSION THREAD

**Discussion by dlvhdr** [dlvhdr/gh-dash#381] [Ideas]
I've been working on this new view of the PR. Unfortunately it requires a lot more work for a full feature parity with github.
If anyone wants to help or move it forward please let me know.

Attached is the design:
![image](https://github.com/dlvhdr/gh-dash/assets/6196971/c97e8ed3-a409-4422-9c5f-643ff806ec78)

This is my WIP:

![image](https://github.com/dlvhdr/gh-dash/assets/6196971/61fba0b3-1d49-44f0-ae91-2ac105e41e1b)

The branch is `pr-view`

**Comment by alephnull** [comment]
This is interesting to me. I checked out `pr-view` and rebased latest main to it. It appears v4 is used so I had to make the following patch
```diff
diff --git a/mocks/mockPr.go b/mocks/mockPr.go
index d87d977..7fdbbb3 100644
--- a/mocks/mockPr.go
+++ b/mocks/mockPr.go
@@ -5,7 +5,7 @@ import (
 
 	graphql "github.com/cli/shurcooL-graphql"
 
-	"github.com/dlvhdr/gh-dash/data"
+	"github.com/dlvhdr/gh-dash/v4/data"
 )
 
 var Pr = data.PullRequestData{
diff --git a/ui/pr.go b/ui/pr.go
index 233bf0e..cdf3461 100644
--- a/ui/pr.go
+++ b/ui/pr.go
@@ -11,13 +11,13 @@ import (
 	tea "github.com/charmbracelet/bubbletea"
 	"github.com/charmbracelet/lipgloss"
 
-	"github.com/dlvhdr/gh-dash/data"
-	"github.com/dlvhdr/gh-dash/mocks"
-	"github.com/dlvhdr/gh-dash/ui/common"
-	"github.com/dlvhdr/gh-dash/ui/markdown"
-	"github.com/dlvhdr/gh-dash/ui/pr"
-	"github.com/dlvhdr/gh-dash/ui/theme"
-	"github.com/dlvhdr/gh-dash/utils"
+	"github.com/dlvhdr/gh-dash/v4/data"
+	"github.com/dlvhdr/gh-dash/v4/mocks"
+	"github.com/dlvhdr/gh-dash/v4/ui/common"
+	"github.com/dlvhdr/gh-dash/v4/ui/markdown"
+	"github.com/dlvhdr/gh-dash/v4/ui/pr"
+	"github.com/dlvhdr/gh-dash/v4/ui/theme"
+	"github.com/dlvhdr/gh-dash/v4/utils"
 )
 
 // Component represents a Bubble Tea model that implements a SetSize function.
diff --git a/ui/pr/common.go b/ui/pr/common.go
index f8c72ad..6a305c7 100644
--- a/ui/pr/common.go
+++ b/ui/pr/common.go
@@ -5,9 +5,9 @@ import (
 
 	"github.com/charmbracelet/log"
 
-	ghctx "github.com/dlvhdr/gh-dash/ui/context"
-	"github.com/dlvhdr/gh-dash/ui/keys"
-	"github.com/dlvhdr/gh-dash/ui/theme"
+	ghctx "github.com/dlvhdr/gh-dash/v4/ui/context"
+	"github.com/dlvhdr/gh-dash/v4/ui/keys"
+	"github.com/dlvhdr/gh-dash/v4/ui/theme"
 )
 
 type Common struct {
@@ -15,7 +15,7 @@ type Common struct {
 	Width, Height int
 	Styles        *ghctx.Styles
 	Theme         *theme.Theme
-	KeyMap        *keys.KeyMap
+	KeyMap        **keys.KeyMap
 	Logger        *log.Logger
 }
 ```
However, when switching PRs, I get a panic. My primary interest is in expanding the PR view to show action runs and run logs. I want to be able to trigger re-runs (all/failed) jobs. Do you have any thoughts about accommodating actions in the design posted above?

This looks like a really involved change and I'm not sure a first-timer like me will be able to contribute meaningfully to the larger goals of this idea to make the view look like the web view so should I base my work off `main`?

**Comment by dlvhdr** [comment]
I worked on https://gh-dash.dev/enhance and it supports viewing logs and rerunning actions.
